### ISSUE TITLE
[release]: python ML library opensearch-py-ml

### ISSUE THREAD

**Issue by ylwu-amzn** [opensearch-project/opensearch-build#2814]
### What is the name of your component?

opensearch-py-ml

### What is the link to your GitHub repo?

https://github.com/opensearch-project/opensearch-py-ml

### Targeted release date

Nov 15

### Where should we publish this component?

pypi

### What type of artifact(s) will be generated for this component?

python library

### Have you completed the required reviews including security reviews, UX reviews?

Security review will be done by Nov 12.

### Have you on-boarded automated security scanning for the GitHub repo associated with this component?

Yes

**Comment by gaiksaya** [comment]
We are in process of automating it so it might fall under `publishToPyPi` library here https://github.com/opensearch-project/opensearch-build/issues/2505

**Comment by ylwu-amzn** [comment]
> We are in process of automating releases so it might fall under `publishToPyPi` library here #2505

Cool, thanks a lot. any work we need do from our side to onboard this auto release workflow?

**Comment by ylwu-amzn** [comment]
@bbarani @gaiksaya For this question "Have you completed the required reviews including security reviews, UX reviews?
",  we are working with security team for security review now. Should we also ask UX team to review considering this a python library?

**Comment by gaiksaya** [comment]
AFAIK, that is for front end plugins/components (if any). @bbarani please correct if I am wrong.

**Comment by bbarani** [comment]
I don’t think we need a UX review until this library has UI. Please update us once the security review is completed CC: @CEHENKLE @elfisher

**Comment by dhrubo-os** [comment]
@bbarani 2 questions:

1. For this library, Security review will be done by 11/12/2022 (ETA given by the reviewer). Will that be a blocker for opensearch-py-ml 2.4 release?
2. Is [this issue](https://github.com/opensearch-project/opensearch-clients/issues/18) mandatory for 2.4 release? As this library is also a client plugin we're going to face the same issue.

**Comment by bbarani** [comment]
Hello @dhrubo-os @ylwu-amzn Does this have to be released along with OpenSearch 2.4 release? As per my understanding, this is a standalone library and can be released independent of 2.4 release. Please let us know if this has to be released on specific time co-relating with any other release.

**Comment by dhrubo-os** [comment]
Hi @bbarani Yeah this is a standalone library and can be release independent of 2.4 release. That being said, this plugin will help customer to simplify of few steps in mlCommons custom model feature, which is why we want to release this plugin asap.

Please suggest what are the steps are remaining from our end to release this plugin. 

Thanks
Dhrubo

**Comment by gaiksaya** [comment]
Hi @dhrubo-os,
I'll update the issue if anything is required from your end.

For now below are the to-do on our list:
1. Get the fix in for https://github.com/opensearch-project/opensearch-build/issues/368 as `.asc` signature is required to sign the packages to PyPi
2. Create the PublishToPyPi jenkins library https://github.com/opensearch-project/opensearch-build/issues/2839
3. Create automation in opensearch-py-ml repo that includes creating a release-drafter workflow, similar to https://github.com/opensearch-project/opensearch-js/blob/main/.github/workflows/release-drafter.yml
4. Set up webhook, secrets and other manual set up so that the jenkins workflow is triggered.

**Comment by gaiksaya** [comment]
The release was successful. Get the artifacts here: https://pypi.org/project/opensearch-py-ml/
